src/dyncfg/src/lib.rs

```diff
@@ -225,6 +221,43 @@ impl ConfigEntry {
     pub fn val(&self) -> ConfigVal {
         self.val.load()
     }
+
+    /// Constructs a handle to the value of this config in the set.
+    ///
+    /// It is the caller's responsibility to construct the handle with the
+    /// correct type.
+    fn handle<T>(&self) -> &ConfigValHandle<T>
+    where
+        T: ConfigType,
+    {
+        ConfigValHandle::<T>::new(&self.val)
+    }
+}
+
+/// A handle to a configuration value in a [`ConfigSet`].
+///
+/// Allows users to amortize the lookup of a name within a set.
+///
+/// Handles can be cheaply cloned.
+#[derive(Debug, Clone, RefCastCustom)]
+#[repr(transparent)]
+pub struct ConfigValHandle<T> {
+    val: ConfigValAtomic,
+    _type: PhantomData<T>,
+}
+
+impl<T> ConfigValHandle<T>
+where
+    T: ConfigType,
+{
+    #[ref_cast_custom]
```

Comment:
I don't love the ref_cast stuff, lots of magic, macros, and third party crates.

Would something like this work instead: keep `Config::shared` but make it not-pub, keep `Config::get` implemented in terms of `shared`, in `Config::handle` clone the `ConfigValAtomic` and use it to construct the handle?

Reply:
Can do, done.